src/components/select/select-list/select-list.component.js

```diff
@@ -93,63 +107,92 @@ const SelectList = React.forwardRef(
       [anchorElement]
     );
 
-    const optionRefList = useMemo(
-      () =>
-        React.Children.map(children, (child) => {
-          if (child?.type === Option || child?.type === OptionRow) {
-            return React.createRef();
-          }
-          return null;
-        }).filter((child) => child),
-      [children]
-    );
-
     const handleSelect = useCallback(
       (optionData) => {
         onSelect({ ...optionData, selectionType: "click" });
       },
       [onSelect]
     );
 
-    const childIds = useMemo(() => React.Children.map(children, () => guid()), [
+    const childIdsRef = useRef(null);
+
+    // childIds should be stable except when children are added or removed - can't use useMemo
+    // as that isn't absolutely guaranteed to never rerun when dependencies haven't changed.
+    const setChildIds = () => {
+      const idArray = [];
+      for (let i = 0; i < children.length; i++) {
+        idArray.push(guid());
+      }
+      childIdsRef.current = idArray;
+    };
+
+    if (
+      childIdsRef.current === null ||
+      childIdsRef.current.length !== React.Children.count(children)
```

Comment:
```suggestion
      if (childIdsRef.current?.length !== React.Children.count(children)) {
```
I think this might be sufficient here